src/kiwixapp.cpp

```diff
@@ -147,14 +147,9 @@ QString KiwixApp::findLibraryDirectory()
     if (isPortableMode())
         return currentDataDir + QDir::separator() + "data";
 
-    // Check for library.xml in the same directory as the executable.
-    auto libraryFile = QFileInfo(currentDataDir, "library.xml");
-    if (libraryFile.exists())
-        return currentDataDir;
-
     // Check for default dataDirectory.
     currentDataDir = getDataDirectory();
```

Comment:
Now, taking into account the current implementation of `getDataDirectory()`, the beginning of this function's body can be simplified as follows:

```c++
{
    auto currentDataDir = getDataDirectory();
    if (isPortableMode())
        return currentDataDir;

   ...
```

Reply:
`getDataDirectory()`'s fallback/final evaluation is the current directory. I do not think this is ideal for the use we need it.

Am I missing something?

Reply:
Maybe I should refactor the variable names which were like this from the libkiwix logic, to have the code be more understandable as well?

Reply:
When `isPortableMode() == true`, `getDataDirectory()` returns the `data` subdirectory of executable's directory, doesn't it?